client/state/sites/vouchers/actions.js

```diff
@@ -103,3 +113,33 @@ export function requestSiteVouchers( siteId ) {
 			} );
 	};
 }
+
+/**
+ * Assign a voucher to the given site.
+ *
+ * @param {Number} siteId - identifier of the site
+ * @param {String} serviceType - service type
+ * @returns {Function} a promise that will resolve once fetching is completed
+ */
+export function assignSiteVoucher( siteId, serviceType ) {
+	return dispatch => {
+		dispatch( vouchersRequestAction( siteId ) );
+
+		return wpcom
+			.site( siteId )
+			.adCreditVouchers()
+			.assign( serviceType )
```

Comment:
Sorry, my bad. Those methods are defined in [wpcom.js](https://github.com/Automattic/wpcom.js/blob/master/lib/site.ad-credit-vouchers.js#L57).